### DISCUSSION TITLE
[Feature] Add an option to sort sharing albums

### DISCUSSION THREAD

**Discussion by sigulete** [immich-app/immich#7025] [Feature Request]
### The feature

I would like to ask if it is possible to add an option to sort albums that are shared by others.

Basically, bringing the existing functionality from my Albums into Sharing. This way, I could sort other people's Albums using the criteria I choose.

![Screenshot from 2024-02-11 17-11-37](https://github.com/immich-app/immich/assets/35115643/4db8534a-c621-4b42-a38c-0747c844c543)

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by Philshappy** [comment]
I agree. Also adding reverse order would be very helpful.

**Reply by martabal** [reply]
There's already a reverse order, you juste have to re-click on the option.

**Comment by Philshappy** [comment]
I am sorry but I don't see it. When I open Immich using my browser and select sharing (left side under map) I can see all my albums that I have shared but there are no options for sort order. Am I missing something.

**Reply by martabal** [reply]
[Screencast from 2024-02-19 19-21-21.webm](https://github.com/immich-app/immich/assets/74269598/bf743b70-824a-473d-a011-c31de4c366c8)

**Comment by Philshappy** [comment]
Thanks. Sorry I cannot see that option when I open a shared album to view the photos in a specific sort. I can only see that option for sorting the albums themselves. When I send out a shared link I would like for the recipient to see the photos inside the album either by newest date or oldest date or by name ascending or descending.

**Reply by martabal** [reply]
Oh yes, it's not yet available for shared albums, I tried to implement it a few months ago but didn't succeed

**Reply by roserez** [reply]
Hi Martin. Since you tried to implement it, your obviously familiar the the relevant areas of the code. Would it be a simple patch to just change the sort to be chronological versus reverse chronological - just hardcoding it to be the opposite of what it is now? Or there many areas involved and things that might break? All of my use cases prefer a chronological sort so I'd be happy to change it across the board as a quick fix. - Mike

**Reply by martabal** [reply]
Hi !
Just to make sure I understand correctly, are you talking about the list of albums in the sharing tab or the order of photos in albums ?

**Reply by roserez** [reply]
The order of the photos within the album. It makes the most sense for people to see events in the order they happened.

**Comment by Philshappy** [comment]
Maybe in the future it will be added.
________________________________

**Comment by Philshappy** [comment]
I think it would be great to have both. The photos within the album being shared should have an option similar to the albums when not being shared, forward or backward. But the list of shared albums should be displayed either by date or by name.


________________________________

**Reply by roserez** [reply]
Yes, I see that there are two feature requests popping up now and again - sorting of Albums and sorting of their contents. Contents is the critical one for me.

**Reply by martabal** [reply]
Yes, that's a totally different FR, I'm pretty sure it has already been requested.

For the original request, I refactored the album list, so it will be much easier to add it for the sharing tab. The PR has been reviewed and is ready to be merged. As soon as it's merged, I'll add it to the sharing tab.

For your FR (change the order of the photos within the album), you can't simply change it with a patch or hard-coding values. I might take a look after implementing this FR.

**Comment by bo0tzz** [comment]
#4716
